model/textparse/promparse.go

```diff
@@ -531,3 +553,40 @@ func parseFloat(s string) (float64, error) {
 	}
 	return strconv.ParseFloat(s, 64)
 }
+
+// baseName returns the metric family name without the _suffix.
+// It uses yoloString for efficiency and avoids unnecessary allocations.
+// This is safe because the returned string is only used during the current parse step
+// and not stored for later use.
+func baseName(b []byte) string {
+	name := yoloString(b)
+	suffixes := []string{"_total", "_count", "_sum", "_bucket"}
+	for _, suffix := range suffixes {
+		if strings.HasSuffix(name, suffix) {
+			return name[:len(name)-len(suffix)]
+		}
+	}
+	return name
+}
+
+func (p *PromParser) handleMetricFamilySwitch() error {
+	metricNameBytes := p.series[p.offsets[0]-p.start : p.offsets[1]-p.start]
+	base := baseName(metricNameBytes)
+
+	// Only do family switch logic when type and unit labels are enabled
+	if p.enableTypeAndUnitLabels {
```

Comment:
Hm we probably want to do it always, ideally. Can we aim for that? We want reliable metadata for various cases, long term, even if it costs ~X% more CPU per parsing.